corehq/apps/data_cleaning/models.py

```diff
@@ -747,6 +739,12 @@ def create_default_filters(cls, session):
                 filter_type=filter_type,
             )
 
+    @classmethod
+    def apply_filters_to_query(cls, session, query):
+        for pinned_filter in session.pinned_filters.all():
+            query = pinned_filter.filter_query(query)
+        return query
+
```

Comment:
Same comment as with `BulkEditFilter`'s refactoring. I'd think this code more naturally lives in the session.

Reply:
I disagree. I want the filter related code to live together, same with columns, etc. I was following the pattern used for columns. If session has too much other model specific logic in it, then I find it difficult to understand tbh